tests/System.IO.Pipelines.Tests/PipelineReaderWriterFacts.cs

```diff
@@ -645,6 +645,64 @@ public void FlushAsync_ReturnsCompletedTaskWhenMaxSizeIfZero()
             Assert.True(flushTask.IsCompleted);
         }
 
+        [Fact]
+        public void TryRead_ShouldReturnTrueOnCommit()
```

Comment:
Yes, `TryRead+TryRead` without advancing. I think all of them should throw. `TryRead + ReadAsync` without advancing too.